internal/datastore/memdb/memdb.go

```diff
@@ -38,32 +37,29 @@ const (
 	indexUsersetNamespace      = "usersetNamespace"
 	indexUsersetRelation       = "usersetRelation"
 	indexUserset               = "userset"
+	indexCreatedTxn            = "createdTxn"
+	indexDeletedTxn            = "deletedTxn"
 
 	defaultWatchBufferLength = 128
 
+	deletedTransactionID = ^uint64(0)
+
 	errUnableToInstantiateTuplestore = "unable to instantiate datastore: %w"
 )
 
-type changelog struct {
-	id         uint64
-	name       string
-	replaces   []byte
-	oldVersion uint64
-}
-
 type namespace struct {
 	name        string
 	configBytes []byte
-	version     uint64
+	createdTxn  uint64
+	deletedTxn  uint64
 }
 
-type tupleChangelog struct {
+type transaction struct {
 	id        uint64
 	timestamp uint64
-	changes   []*v0.RelationTupleUpdate
 }
 
-type tupleEntry struct {
+type relationship struct {
```

Comment:
Since this is being renamed to relationship, thoughts on renaming the members to resource/subject?